### PR TITLE
consume waka api

### PR DESCRIPTION
**Pull request by jvelezpo**
* updates electron version to 9
* consumes summaries API to show daily coding time in tray menu
* use export default for new util classes
* shows in tray menu the amount of time spend in the day
image

### CODE REVIEW

File: src/utils/api.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import Options from './options';
+
+const options = new Options();
+export default class API {
+  public async summaries() {
+    try {
+      const apikey = await options.getApiKeyAsync();
+      const token = Buffer.from(apikey).toString('base64');
+      axios.defaults.baseURL = 'https://wakatime.com';
```
Comment (gandarez) [inline src/utils/api.ts:11]: What about define it as constant `const baseURL = 'https://wakatime.com/api/'` ?

File: src/utils/api.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      axios.defaults.headers.common.Authorization = `Basic ${token}`;
+      const response = await axios.get('/api/v1/users/current/summaries?start=today&end=today');
```
Comment (gandarez) [inline src/utils/api.ts:13]: if baseURL becomes a constant then we need to remove `api` from the url.

File: src/utils/api.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import axios from 'axios';
+
+import Options from './options';
+
+const options = new Options();
+export default class API {
+  public async summaries() {
```
Comment (gandarez) [inline src/utils/api.ts:7]: I believe making this method to accept `start date and end date` would turn it more flexible for future implementations.

File: src/utils/api.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        return null;
+      }
+      return response.data;
+    } catch (err) {
+      return null;
+    }
+  }
+
+  public async todayMins() {
+    const summaries = await this.summaries();
```
Comment (gandarez) [inline src/utils/api.ts:24]: Might call `this.summaries()` passing `start date and end date`?

File: src/utils/api.ts (hunk 1 above)
Comment (jvelezpo) [reply src/utils/api.ts]: i wanted to go as the wakatime docs shows for ecample look https://wakatime.com/developers#summaries
all the endPoints include the `api`, thats why i dont include it in the baseUrl
`GET` `/api/v1/users/:user/summaries`
`GET` `/api/v1/users/current/summaries`

File: src/utils/api.ts [hunk 5]
```diff
@@ -3,14 +3,16 @@ import axios from 'axios';
 import Options from './options';
 
 const options = new Options();
+const baseURL = 'https://wakatime.com';
```
Comment (gandarez) [inline src/utils/api.ts:6]: Because it's in the modeule `API` let's make it ending with `/api`.

Comment (jvelezpo) [reply src/utils/api.ts:6]: done

**Pull request merged**